### PR TITLE
Check file size against memory limit prior to attempting a forced download #3310

### PR DESCRIPTION
**Pull request by cklosowski**
#3310

### CODE REVIEW

File: includes/process-download.php [hunk 1]
```diff
@@ -233,13 +233,25 @@ function edd_process_download() {
 
 				}
 
+				// Determine if the file we're attempting to use direct on is too large for the memory limit to use direct
+				$memory_limit = @ini_get( 'memory_limit' );
```
Comment (pippinsplugins) [inline includes/process-download.php]: This should probably only happen if `$direct` is `true`

File: includes/process-download.php [hunk 2]
```diff
@@ -233,13 +233,25 @@ function edd_process_download() {
...
+				if ( $memory_limit ) {
+					$memory_limit_bytes = edd_determine_size_in_bytes( $memory_limit );
+					$file_size          = edd_determine_size_in_bytes( @filesize( ABSPATH . $file_path ) );
```
Comment (pippinsplugins) [inline includes/process-download.php]: Above we do `@filesize( $file_path )` so `ABSPATH` shouldn't be used here I don't think.

File: includes/process-download.php [hunk 3]
```diff
@@ -838,3 +850,23 @@ function edd_set_requested_file_scheme( $requested_file, $download_files, $file_
 
 }
 add_filter( 'edd_requested_file', 'edd_set_requested_file_scheme', 10, 3 );
+
+/**
+ * Given a string for a file size, dtermeine the size in bytes
```
Comment (michaelbeil) [inline includes/process-download.php]: s/dtermeine/determine

Comment (sunnyratilal) [inline includes/process-download.php]: Fixed

File: includes/process-download.php (hunk 2 above)
Comment (sunnyratilal) [inline includes/process-download.php]: Yep `ABSPATH` isn't required here

Comment (sunnyratilal) [inline includes/process-download.php]: Fixed

File: includes/process-download.php (hunk 1 above)
Comment (sunnyratilal) [inline includes/process-download.php]: Yep, fixed it up.

**Pull request closed without merging**